src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferStringContainsOverIndexOf.Fixer.cs

```diff
@@ -0,0 +1,144 @@
+                    {
+                        DocumentEditor editor = await DocumentEditor.CreateAsync(document, cancellationToken).ConfigureAwait(false);
+                        SyntaxEditor syntaxEditor = new SyntaxEditor(treeRoot, document.Project.Solution.Workspace);
+                        SyntaxGenerator generator = editor.Generator;
+                        var containsExpression = generator.MemberAccessExpression(localReferenceOrParameterNode, "Contains");
+                        List<SyntaxNode> stringComparisonArguments = new List<SyntaxNode>();
+                        SyntaxNode? containsInvocation = null;
+                        int numberOfArguments = indexOfMethodArguments.Length;
+                        if (numberOfArguments == 1)
+                        {
+                            var firstArgument = indexOfMethodArguments.First();
+                            if (firstArgument.Parameter.Type.Equals(charType))
+                            {
+                                containsInvocation = generator.InvocationExpression(containsExpression, firstArgument.Syntax);
+                            }
+                            else
+                            {
+                                stringComparisonArguments.Add(firstArgument.Syntax);
+                                var systemNode = generator.IdentifierName("System");
+                                var argument = generator.MemberAccessExpression(generator.MemberAccessExpression(systemNode, "StringComparison"), "CurrentCulture");
+                                containsInvocation = generator.InvocationExpression(containsExpression, firstArgument.Syntax, argument);
+                            }
+                        }
+                        else
+                        {
+                            IArgumentOperation secondArgument = indexOfMethodArguments[1];
+                            if (secondArgument.Value.ConstantValue.HasValue && secondArgument.Value.ConstantValue.Value is int intValue)
+                            {
+                                if (intValue == (int)StringComparison.Ordinal)
+                                {
+                                    containsInvocation = generator.InvocationExpression(containsExpression, indexOfMethodArguments[0].Syntax);
+                                }
+                                else
+                                {
+                                    containsInvocation = generator.InvocationExpression(containsExpression, indexOfMethodArguments[0].Syntax, indexOfMethodArguments[1].Syntax);
+                                }
+                            }
+                        }
+                        SyntaxNode newIfCondition = containsInvocation!;
```

Comment:
Bug here: containsInvocation can still be null here if the `StringComparison` argument is not a constant value but rather a variable